Issue 4: Add /chat/{id}/entry API call

YuliaMittova opened the pull request:
Send message to LLM and print response in chat #4
It should be possible to send a request to /chat/{id}/entry with a prompt param
this message should be saved with role USER
this message should be sent into LLM chat service
The response from LLM should be saved into database with role ASSISTANT
FAIL_TO_PASS: com.example.llmchat.integration.ChatEntryApiIntegrationTest

YuliaMittova commented:
@issue-validator validate the issue

YuliaMittova committed:
parent c8878fbb4e4464375b3614959c4e7c77cac9b13c
author ymittova <kostyukhina.julia@gmail.com> 1766532973 +0100
committer ymittova <kostyukhina.julia@gmail.com> 1766751023 +0100

Create database entities to store chat history #1
Create the following hibernate entities:
1. Chat(id, title, createdAt, history (which contains all chatEntities)
2. ChatEntity(id, role, createdAt) Role is an enum with the following possible values (user, assistant, system).
Each enum should have the name and implement the method abstract fun getMessage(prompt: String): Message, where message is from org.springframework.ai.chat.messages
FAIL_TO_PASS: com.example.llmchat.repository.ChatPersistenceTest, com.example.llmchat.model.RoleTest

YuliaMittova commented:
@issue-validator validate the issue

YuliaMittova commented:
@issue-validator validate the issue

YuliaMittova commented:
@issue-validator validate the issue

YuliaMittova commented:
@issue-validator validate the issue

YuliaMittova commented:
@issue-validator validate the issue

YuliaMittova commented:
@issue-validator validate the issue

DmitriyChernyakSetronica commented:
@issue-validator validate the issue

DmitriyChernyakSetronica commented:
green validation is https://github.com/dpaia/infrastructure/actions/runs/20528810131/job/58994500690

YuliaMittova committed:
Send message to LLM and print response in chat #4
It should be possible to send a request to /chat/{id}/entry with a prompt param
this message should be saved with role USER
this message should be sent into LLM chat service
The response from LLM should be saved into database with role ASSISTANT
FAIL_TO_PASS: com.example.llmchat.integration.ChatEntryApiIntegrationTest

YuliaMittova commented:
@issue-validator validate the issue

DmitriyChernyakSetronica commented:
@issue-validator validate the issue

DmitriyChernyakSetronica commented:
ChatEntryApiIntegrationTest doesn't check for saving to the database - doesn't check requirements 2 and 4

DmitriyChernyakSetronica commented on src/test/kotlin/com/example/llmchat/integration/ChatEntryApiIntegrationTest.kt:
```diff
@@ -0,0 +1,188 @@
+
+    @Test
+    fun `when unknown chat id is passed then 404 error is returned`() {
+        val url = "http://localhost:$port/chats/999/entry?prompt=hi"
```
url should be /chat/999/entry not /chats/999/entry

YuliaMittova replied:
Fixed in the same commit

DmitriyChernyakSetronica commented on src/test/kotlin/com/example/llmchat/integration/ChatEntryApiIntegrationTest.kt:
```diff
@@ -0,0 +1,188 @@
+        assertTrue(answer.contains(ANSWER))
+
+        wiremock.verify(postRequestedFor(urlEqualTo("/api/chat")))
+    }
```
To check requirements 2 and 4, for example, the agent suggested adding the following block
     
  // Then: Verify messages were saved to database (requirements 2 and 4)
        val chat = chatRepository.findById(chatId).orElseThrow()
        assertTrue(chat.history.size >= 2, "Should have at least USER and ASSISTANT messages")
        
        // Find the messages we just added (last 2 messages)
        val userMessage = chat.history.findLast { it.role == Role.USER && it.content == prompt }
        val assistantMessage = chat.history.findLast { it.role == Role.ASSISTANT && it.content.contains(ANSWER) }
        
        assertNotNull(userMessage, "USER message should be saved to database")
        assertNotNull(assistantMessage, "ASSISTANT message should be saved to database")
        assertEquals(prompt, userMessage.content, "USER message content should match prompt")
        assertTrue(assistantMessage.content.contains(ANSWER), "ASSISTANT message should contain LLM response")
    }

YuliaMittova replied:
Fixed in the same commit.

DmitriyChernyakSetronica commented:
please add test case for "when LLM returns empty response then 502 error is returned"
After this, check the full coverage of integration tests and it is worth removing all similar test units.
I believe this ChatEntryApiIntegrationTest test  alone will be sufficient.

YuliaMittova commented:
> please add test case for "when LLM returns empty response then 502 error is returned" After this, check the full coverage of integration tests and it is worth removing all similar test units. I believe this ChatEntryApiIntegrationTest test alone will be sufficient.

Added this test case to the integration test. As far as I know as per test pyramid integration tests are not replacing unit tests. But if you want I can delete them.

DmitriyChernyakSetronica commented:
@issue-validator validate the issue

DmitriyChernyakSetronica commented:
@issue-validator validate the issue

DmitriyChernyakSetronica commented:
Verified, accepted

kaleev commented:
Please extract the addition of test dependencies into a separate base commit.
